libraries/chain/combined_database.cpp

```diff
@@ -161,30 +161,30 @@ namespace eosio { namespace chain {
        : backing_store(backing_store_type::ROCKSDB), db(chain_db), kv_database{ [&]() {
             rocksdb::Options options;
 
-            options.create_if_missing = !cfg.read_only; // Creates a database if it is missing
+            options.create_if_missing = true; // Creates a database if it is missing
             options.level_compaction_dynamic_level_bytes = true;
-            options.bytes_per_sync = 1048576; // used to control the write rate of flushes and compactions.
+            options.bytes_per_sync = cfg.persistent_storage_bytes_per_sync; // used to control the write rate of flushes and compactions.
 
             // By default, RocksDB uses only one background thread
             // for flush and compaction.
             // Good value for `total_threads` is the number of cores
-            options.IncreaseParallelism(cfg.rocksdb_threads);
+            options.IncreaseParallelism(cfg.persistent_storage_num_threads);
 
             options.OptimizeLevelStyleCompaction(512ull << 20); // optimizes level style compaction
 
             // Number of open files that can be used by the DB.  
             // Setting it to -1 means files opened are always kept open.
-            options.max_open_files = cfg.rocksdb_max_open_files;
+            options.max_open_files = cfg.persistent_storage_max_num_files;
 
             // Use this option to increase the number of threads
             // used to open the files.
-            options.max_file_opening_threads = cfg.rocksdb_threads; // Default should be the # of Cores
+            options.max_file_opening_threads = cfg.persistent_storage_num_threads; // Default should be the # of Cores
```

Comment:
@victorj8 - we should remove the comment on this line as it applied to the earlier setting and is confusing.

Reply:
Done